### ISSUE TITLE
Binary matrix factorization input data format issue

### ISSUE THREAD

**Issue by Alex-richardson1** [yixuan/recosystem#13]
For binary matrix factorisation, the input matrix only contains ratings equal to 1, since in sparse matrices, the vast majority of entries in the matrix will be empty, and thus these should not be included in the training data.

However, the recommender$train function only seems to work if the training matrix contains some -1 values in the rating column as well, otherwise nans are returns for every training iteration. Is this expected behaviour ?  The point of the sparse matrix triplet form is to only include non-zero entries, I would think.

**Comment by yixuan** [comment]
I think what you meant was the "one-class matrix", rather than the "binary matrix". Have you specified `opts = list(loss = "row_log")` or "opts = list(loss = "col_log")" in `$train()`?

**Comment by Alex-richardson1** [comment]
Hi. Yes, I meant one-class matrix, apologies.    I get the following error if I include row_log or col_log in the opts parameter :   "Error in recommender$train(train_data, opts = c(dim = 30, costp_l2 = 0.1,  : 
  Not compatible with requested type: [type=character; target=double]."

**Comment by Alex-richardson1** [comment]
Actually I just realised the cause of my errors,  I was entering the options as   opts = c( parameters) as opposed to  opts = list(parameters).  Seems to work now.    Thanks
